src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -69,6 +69,7 @@ internal partial class Binder
                     // x => ...
                     hasSignature = true;
                     var simple = (SimpleLambdaExpressionSyntax)syntax;
+                    ReportFieldOrValueContextualKeywordConflictIfAny(simple.Parameter, simple.Parameter.Identifier, diagnostics);
```

Comment:
>ReportFieldOrValueContextualKeywordConflictIfAny [](http://example.com/codeflow?start=20&length=48)

Are we missing a check the ```case SyntaxKind.AnonymousMethodExpression``` code path? Perhaps instead of performing a check inside ```CheckParenthesizedLambdaParameters``` we should perform it below, when we are processing ```parameterSyntaxList``` and checking for underscores used as identifiers for parameters.